improver/utilities/pad_spatial.py

```diff
@@ -219,23 +219,17 @@ def pad_cube_with_halo(cube, width_x, width_y, halo_mean_data=True):
 
     # Pad a halo around the original data with the extent of the halo
     # given by width_y and width_x.
-    if halo_mean_data:
-        padded_data = np.pad(
-            cube.data,
-            ((width_y, width_y), (width_x, width_x)),
-            "mean",
-            stat_length=(
-                (0.5 * width_y, 0.5 * width_y),
-                (0.5 * width_x, 0.5 * width_x),
-            ),
-        )
-    else:
-        padded_data = np.pad(
-            cube.data,
-            ((width_y, width_y), (width_x, width_x)),
-            "constant",
-            constant_values=(0.0, 0.0),
-        )
+    kwargs = {
+        "stat_length": ((width_y // 2, width_y // 2), (width_x // 2, width_x // 2))
+    }
+    if pad_method == "constant":
+        kwargs = {"constant_values": (0.0, 0.0)}
+    if pad_method == "symmetric":
+        kwargs = {}
+
+    padded_data = np.pad(
+        cube.data, ((width_y, width_y), (width_x, width_x)), pad_method, **kwargs
```

Comment:
As `pad_method` is a keyword argument, it might be slightly better to do: `mode=pad_method`.

Reply:
Agreed. Changed.